packages/next-codemod/transforms/next-request-geo-ip.ts

```diff
@@ -0,0 +1,451 @@
+import type {
+  API,
+  ASTPath,
+  FileInfo,
+  FunctionDeclaration,
+  Collection,
+  Node,
+  ImportSpecifier,
+  Identifier,
+} from 'jscodeshift'
+
+const GEO = 'geo'
+const IP = 'ip'
+const GEOLOCATION = 'geolocation'
+const IP_ADDRESS = 'ipAddress'
+const GEO_TYPE = 'Geo'
+
+export default function (fileInfo: FileInfo, api: API) {
+  const j = api.jscodeshift.withParser('tsx')
+  const ast = j(fileInfo.source)
```

Comment:
Align the name `root` with others, since `j()` will always return a collection, which represents the ast path, this is the special one cause it's the ast path of whole file
```suggestion
  const root = j(fileInfo.source)
```